### PR TITLE
Kelsha Darby - User Story 45

### LINKED ISSUE #48: User Story 45, Merchant adds an item
As a merchant employee
When I visit my items page
I see a link to add a new item
When I click on the link to add a new item
I see a form where I can add new information about an item, including:
- the name of the item, which cannot be blank
- a description for the item, which cannot be blank
- a thumbnail image URL string, which CAN be left blank
- a price which must be greater than $0.00
- my current inventory count of this item which is 0 or greater

When I submit valid information and submit the form
I am taken back to my items page
I see a flash message indicating my new item is saved
I see the new item on the page, and it is enabled and available for sale
If I left the image field blank, I see a placeholder image for the thumbnail

### PR DESCRIPTION
**Pull request by kelshadarby**
## A merchant user can create an item with or without an image
Closes #48 

**_Test_**
- Created test to check that a merchant user can create an item with or without filling in the image field

**_Routes_**
- Add merchant item creation routes

**_Controller_**
- Added a `create` method to the merchant items controller

**_Model_**
- Added a `before_save` method to the model that sets the image to its default if it is not filled in in the form

**_Database_**
- Add default value to item image column

### CODE REVIEW

File: spec/features/merchant/items/new_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
```
Comment (brian-greeson) [inline spec/features/merchant/items/new_spec.rb:2]: Nice test

**Review by brian-greeson — APPROVED** [review]

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+  end
+
+  def new
+  end
+
+  def create
+    merchant = Merchant.find(current_user.merchant.id)
+    item = merchant.items.new(item_params)
+    if item.save
+      flash[:success] = "#{item.name} has been created."
```
Comment (fredrondina) [inline app/controllers/merchant/items_controller.rb:14]: nice create method

File: app/models/item.rb [hunk 3]
```diff
@@ -54,4 +53,8 @@ def deletable?
     !orders.any?
   end
 
+  def set_image
+    self.image = "https://upload.wikimedia.org/wikipedia/commons/1/15/No_image_available_600_x_450.svg" if self.image.blank?
```
Comment (fredrondina) [inline app/models/item.rb:57]: great method

**Review by fredrondina — APPROVED** [review]
looks great, nice work

**Pull request merged**